backend/dataall/modules/dataset_sharing/services/share_item_service.py

```diff
@@ -101,16 +102,10 @@ def revoke_items_share_object(uri, revoked_uris):
 
             share_sm.update_state(session, share, new_share_state)
 
-            if share.groupUri != dataset.SamlAdminGroupName:
-                revoke_table_items = ShareObjectRepository.find_all_share_items(
-                    session, uri, ShareableType.Table.value, [ShareItemStatus.Revoke_Approved.value]
-                )
-                for item in revoke_table_items:
-                    ResourcePolicyService.delete_resource_policy(
-                        session=session,
-                        group=share.groupUri,
-                        resource_uri=item.itemUri,
-                    )
+            if share.groupUri != dataset.SamlAdminGroupName and share.principalType == PrincipalType.Group.value:
+                log.info('Deleting TABLE/FOLDER READ permissions...')
+                ShareItemService._delete_dataset_table_read_permission(session, share)
```

Comment:
For tables - I believe there may be a subset of table read permissions associate to CR shares that will never get deleted now  (maybe something we should handle in migration script?)

Reply:
That is what I thought, but surprise surprise there is only one permission that gets created for each group. If a group creates shares with 3 CRs there will only be one permission

Reply:
Okay I see - so for all new CR shares the share requestor Group will not `DATASET_TABLE_READ` (unless they also have access to the table) 

But if there were pre-existing table item shares the  requestor Group would have `DATASET_TABLE_READ`. 

I guess I am just  wondering if we should enforce it one way or the other - but the imapct is minimal i believe

Reply:
After doing some testing - I also see the same can be the case of an approval failure (i.e. `Share_Failed), where:

- Share Approved
- Requestor Group gets Permissions entry in DB for DatasetStorageLocation
- Share Failed --> Only option is to delete share
- Now if approver tries to approve again and all works this is fine (no dups) but at this moment the permission entry is orphaned
  - Requestor Group can still get folder even though Share_Failed

Reply:
Yes, I think we need to do some thinking on 1. lifecycle of "SHARE" permissions 2. duplicates on share permissions. I did not want to add anything to this PR to reduce the scope of the testing. I have created [one issue](https://github.com/data-dot-all/dataall/issues/1173) for the lifecycle of permissions and [another issue](https://github.com/data-dot-all/dataall/issues/1174) for the duplicates coming form consumption roles or groups in multiple environments.